synapse/handlers/message.py

```diff
@@ -640,6 +640,12 @@ async def create_event(
             auth_event_ids=auth_event_ids,
             depth=depth,
         )
+        logger.info(
+            "create_new_client_event %s event=%s state_group=%s",
+            event.type,
+            event.event_id,
+            context._state_group,
+        )
```

Comment:
Maybe can use the same `old_state` trick from https://github.com/matrix-org/synapse/pull/10948 but only do it for `insertion` events which are the root ancestor of each batch.

(also make sure to make the state events as outliers again)